PMSCore/ReportManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using PMSCore;
+
+namespace PMSCore
+{
+    /// <summary>
+    /// This class handles the creation and management of reports.
+    /// </summary>
+    public sealed class ReportManager
+    {
+        // The single instance of ReportManager
+        private static ReportManager _instance;
+
+        // Logger instance
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Private constructor to prevent external instantiation.
+        /// </summary>
+        private ReportManager()
```

Comment:
Pass an ILogger as a parameter if null call on the factory please see how the other Managers were implemented

Reply:
@DistroRickDev , this should be done now, can you please double check?